src/api/java/com/minecolonies/api/research/GlobalResearch.java

```diff
@@ -0,0 +1,267 @@
+        final ILocalResearch localResearch = localTree.getResearch(this.getBranch(), this.getId());
+
+        return localResearch == null && canDisplay(uni_level) && (parentResearch == null || localParentResearch != null && localParentResearch.getState() == ResearchState.FINISHED) && ( parentResearch == null || !parentResearch.hasResearchedChild(localTree) || !parentResearch.hasOnlyChild()) && (depth < 6 || !localTree.branchAlreadyResearchedLevelSix(branch));
+    }
+
+    @Override
+    public boolean canDisplay(final int uni_level)
+    {
+        return uni_level >= depth;
+    }
+
+    @Override
+    public void loadCostFromConfig()
+    {
+        costList.clear();
+        try
+        {
+            final CommonConfiguration configuration = MinecoloniesAPIProxy.getInstance().getConfig().getCommon();
+            final ForgeConfigSpec.ConfigValue<List<? extends String>> researchCost = (ForgeConfigSpec.ConfigValue<List<? extends String>>) configuration.getClass().getDeclaredField(id).get(configuration);
+            for (final String cost : researchCost.get())
+            {
+                final String[] tuple = cost.split("\\*");
+                final Item item = ForgeRegistries.ITEMS.getValue(new ResourceLocation(tuple[0]));
+                if (item == null)
+                {
+                    Log.getLogger().warn("Couldn't retrieve research costList from config for " + branch + "/" + id + " for item: " + tuple[0]);
+                    return;
+                }
+                costList.add(new ItemStorage(new ItemStack(item, 1), Integer.parseInt(tuple[1]), false));
+            }
+        }
+        catch (final NoSuchFieldException | IllegalAccessException | NumberFormatException e)
+        {
+            Log.getLogger().warn("Couldn't retrieve research costList from config for " + branch + "/" + id + " !", e);
+        }
+    }
+
+    @Override
+    public boolean hasEnoughResources(final IItemHandler inventory)
+    {
+        for (final ItemStorage cost: costList)
+        {
+            final int count = InventoryUtils.getItemCountInItemHandler(inventory, stack -> !ItemStackUtils.isEmpty(stack) && stack.isItemEqual(cost.getItemStack()));
+            if (count < cost.getAmount())
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public List<ItemStorage> getCostList()
+    {
+        return ImmutableList.copyOf(costList);
+    }
+
+    @Override
+    public void startResearch(@NotNull final PlayerEntity player, @NotNull final LocalResearchTree localResearchTree)
+    {
+        if (localResearchTree.getResearch(this.branch, this.id) == null && hasEnoughResources(new InvWrapper(player.inventory)))
+        {
+            final ILocalResearch research = new LocalResearch(this.id, this.branch, this.depth);
+            research.setState(ResearchState.IN_PROGRESS);
+            localResearchTree.addResearch(branch, research);
```

Comment:
should we mark the colony dirty here to save the resource consume?

Reply:
We mark the colony dirty after we call this in the message.